### PR TITLE
Improve on-premise runners section

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Please refer to the commit list and messages for more information.

***

> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: content/docs/self-hosted-runners.md [hunk 1]
```diff
@@ -202,16 +202,23 @@ env:
...
+  --token="$personal_access_token" \
+  --labels="custom,runner" \
+  --idle-timeout=180
 ```
 
-Now your machine will be listening for workflows from your project repository.
+Now your machine will listen for jobs triggered on your repository and execute
+them locally.
+
+:warning: **Warning:** people with access to your repository (everybody for
```
Comment (jorgeorpinel) [inline content/docs/self-hosted-runners.md:219]: > `:warning:`

Does that render correctly? 😮 I didn't know that, I usually just paste the actual emoji (also works).

File: content/docs/self-hosted-runners.md [hunk 2]
```diff
@@ -202,16 +202,23 @@ env:
...
+public ones) could execute arbirary code on your machine; please refer to the
+pertaining
```
Comment (jorgeorpinel) [inline content/docs/self-hosted-runners.md:221]:
```suggestion
corresponding
```

File: content/docs/self-hosted-runners.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/self-hosted-runners.md:219]: Nope, that doesn't work 😏 

image

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:219]:
```suggestion
⚠️ **Warning:** people with access to your repository (everybody for
```

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md]: I had to try! 😄  GitHub-flavoured Markdown usually accepts the colon notation. 🤷🏼‍♂️

**Review by jorgeorpinel — APPROVED** [review]
I feel OK to approve this since the changes are mostly to language not so much on CML details.

File: content/docs/self-hosted-runners.md (hunk 1 above)
Comment (jorgeorpinel) [reply content/docs/self-hosted-runners.md]: GH does render it tho 😆 

image

File: content/docs/self-hosted-runners.md [hunk 3]
```diff
@@ -202,22 +202,22 @@ env:
...
 machine or on-premise GPU cluster.
 
 ```dvc
 cml-runner \
   --repo="$repository_url" \
   --token="$personal_access_token" \
-  --labels="custom,runner" \
-  --idle-timeout=180
+  --labels="local,runner" \
+  --idle-timeout=0
```
Comment (casperdcl) [inline content/docs/self-hosted-runners.md:213]: seems odd to have `0`. Maybe leave out?

Comment (casperdcl) [reply content/docs/self-hosted-runners.md:213]: I remember bringing this up before. Really seems like a slightly broken CLI API - should maybe fix in https://github.com/iterative/cml?

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:213]: https://github.com/iterative/cml/pull/700

Comment (casperdcl) [reply content/docs/self-hosted-runners.md:213]: ok but still

```suggestion
```

the idea is that the default is computed and probably sensible (before and after https://github.com/iterative/cml/pull/700)

Comment (casperdcl) [reply content/docs/self-hosted-runners.md:213]: continuing in #93

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:213]: The default value is intended for cloud runners; please refer to the commit message body on 2cbaafaa2a57b055faa1fb274000fc6d49f793e2

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:213]: When running this locally, users (read: myself at least) would probably want the runner to keep listening forever and not to die after 300 seconds of inactivity.

Comment (DavidGOrtega) [reply content/docs/self-hosted-runners.md:213]: > The default value is intended for cloud runners; 

Who says 🙂 ? My own gpus stays until there is not more jobs in the pipeline. Then I shut down the machine via script. Its very energy unfriendly.

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:213]: > _Who says 🙂 ? My own gpus stays until there is not more jobs in the pipeline. Then I shut down the machine via script. Its very energy unfriendly._

Alright! 😄  To my mind, it makes more sense for cloud runners than for on-premises machines, but your approach seems excellent both for the planet and for the budget.

What if we recommend no timeput by default and add a callout bellow?

> 🌳  Note: you can set `--idle-timeout=<seconds>` so `cml runner` exits after `<seconds>` of inactivity, and, perhaps, configure your computer to shutdown automatically afterwards

Comment (DavidGOrtega) [reply content/docs/self-hosted-runners.md:213]: Yeah suggesting 0 sounds good to me

File: content/docs/self-hosted-runners.md [hunk 4]
```diff
@@ -202,16 +202,23 @@ env:
...
 
 ```dvc
 cml-runner \
-  --repo $your_project_repository_url \
-  --token=$personal_access_token \
-  --labels tf \
-  --idle-timeout 180
+  --repo="$repository_url" \
+  --token="$personal_access_token" \
+  --labels="local,runner" \
```
Comment (DavidGOrtega) [inline content/docs/self-hosted-runners.md:212]: I do not see the need of runner tag

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:212]: I added it to illustrate the use of comma-separated tags with no spaces between them, but naming the second tag `runner` is really redundant.

Comment (0x2b3bfa0) [reply content/docs/self-hosted-runners.md:212]: Feel free to create a follow-up pull request removing it.

**Pull request merged**